connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/RetryWithToleranceOperator.java

```diff
@@ -175,9 +179,8 @@ protected <V> V execAndRetry(Operation<V> operation) throws Exception {
                 log.trace("Caught a retriable exception while executing {} operation with {}", context.stage(), context.executingClass());
                 errorHandlingMetrics.recordFailure();
                 if (checkRetry(startTime)) {
-                    backoff(attempt, deadline);
-                    if (Thread.currentThread().isInterrupted()) {
-                        log.trace("Thread was interrupted. Marking operation as failed.");
+                    if (!backoff(attempt, deadline) || Thread.currentThread().isInterrupted()) {
+                        log.trace("Thread was interrupted or exit condition was triggered. Marking operation as failed.");
```

Comment:
Nit: "exit condition was triggered" is pretty vague. What about "shutdown has been scheduled"?